op-service/txmgr/txmgr.go

```diff
@@ -418,46 +427,67 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txHash common.Hash,
 	return nil
 }
 
-// increaseGasPrice takes the previous transaction & potentially clones then signs it with a higher tip.
-// If the tip + basefee suggested by the network are not greater than the previous values, the same transaction
-// will be returned. If they are greater, this function will ensure that they are at least greater by 15% than
-// the previous transaction's value to ensure that the price bump is large enough.
-//
-// We do not re-estimate the amount of gas used because for some stateful transactions (like output proposals) the
-// act of including the transaction renders the repeat of the transaction invalid.
```

Comment:
I was confused by the above comment: the fact that a tx can become invalid seems a reason why you *should* re-estimate gas here, as you detect the tx is invalid earlier before even trying to execute it.

Reply:
the tx is invalidated by itself. Specifically if the proposal transaction is included on chain, but the confirmation depth has not been hit, replaying the transaction in the pending state will alway fail because the output has already been proposed.

Reply:
and in this case re-estimating gas would return an error and abort the replay, which seems the desirable outcome? (The original tx should eventually confirm and then progress will resume from there.)

Reply:
oh I see, the error could cause the original confirmation wait loop to abort before reaching confirmation & reading the receipt, resulting in an unnecessary re-submision by the proposer?

Reply:
Oh and indeed I do see this happen in our logs, once in the last 24 hours. I think all we need to do here is to continue looping waiting on the previous tx should increaseGasPrice error out.